### DISCUSSION TITLE
Feyncalc Error "could not open QCD/QCD.gen"

### DISCUSSION THREAD

**Discussion by ZAarfi** [FeynCalc/feyncalc#218] [Q&A]
Dear All

I am running the example file given in feyncalc examples directory named Gl-GlGl. When I run the code I get the error "InitializeModel::noopen: Cannot open QCD\QCD.gen." I made a manual search for the mentioned QCD.gen file but it is not there in any of the directories of feyncalc. Then there is a long list of errors appearing in the same file. I have attached the notebook saved as pdf that might help. Any help in this regard will be highly appreciated. Thank you. 
[Gl-GlGl error.pdf](https://github.com/FeynCalc/feyncalc/files/11187922/Gl-GlGl.error.pdf)

**Comment by vsht** [comment]
Hi,

did you read the text coming right after "Load FeynCalc and the 
necessary add-ons or other packages"? If yes, did you install FeynRules 
and generate the model?



Am 10.04.23 um 07:32 schrieb ZAarfi:

**Reply by ZAarfi** [reply]
Hi 
Thank you for writing an answer. Yes for your first question, everything loads finely. For your second question I think QCD model comes built in with the feyncalc as some other files in the directory run properly. I am not sure or aware if I have to build QCD model using feynrules. Please guide me where I am making a mistake.

**Reply by vsht** [reply]
Can you please just follow the steps in the above mentioned text? I.e. install FeynRules and run the script mentioned there. If those steps would not be necessary to run that particular example, there would be no point to have that text in the first place.

**Reply by pengfei-physics** [reply]
Hi​
I have the same issue. I found FeynRules and tried to run GenerateModelQCD.m, but it resulted in the error: Get::noopen: Cannot open FeynRules'. Additionally, it warned me with StringJoin::string: $Failed<> it should be a string in position one. Then, I went back to Gl-GlGl and ran it, but it also failed and did not produce any FeynArts output.
![屏幕截图 2025-03-10 202519](https://github.com/user-attachments/assets/1b6e6bba-ac5a-49e1-882d-85d524a77bf7)
![屏幕截图 2025-03-10 202625](https://github.com/user-attachments/assets/0193a232-2098-4998-a1ae-df4921963e52)
![屏幕截图 2025-03-10 202652](https://github.com/user-attachments/assets/86ac5177-b028-4215-9060-f14e49816371)
![屏幕截图 2025-03-10 202722](https://github.com/user-attachments/assets/18933a0c-b7dc-4b82-8264-9f72aca83581)
![image](https://github.com/user-attachments/assets/1f6b4e19-9b2a-471b-81ba-a4ddf08a9cf8)
